internal/backend/runtime/omni/controllers/omni/config_patch_cleanup.go

```diff
@@ -171,54 +172,59 @@ func (ctrl *ConfigPatchCleanupController) initDefaults() {
 	}
 }
 
-// isOrphan checks if the ConfigPatch is orphaned.
-func (ctrl *ConfigPatchCleanupController) isOrphan(ctx context.Context, r controller.Reader, configPatch *omni.ConfigPatch) (bool, error) {
-	if configPatch.Metadata().Owner() != "" {
-		return false, nil
+// isOrphanByLabel returns true if the resource is orphaned by label and linked resource T.
+func isOrphanByLabel[T generic.ResourceWithRD](ctx context.Context, r controller.Reader, label string, configPatch *omni.ConfigPatch) (bool, error) {
+	value, ok := configPatch.Metadata().Labels().Get(label)
+	if !ok {
+		return true, nil
 	}
 
-	if time.Since(configPatch.Metadata().Updated()) < ctrl.DeleteOlderThan {
-		return false, nil
+	_, err := safe.ReaderGetByID[T](ctx, r, value)
+	if err != nil && state.IsNotFoundError(err) {
+		return true, nil
 	}
 
-	if machine, ok := configPatch.Metadata().Labels().Get(omni.LabelMachine); ok {
-		_, err := safe.ReaderGetByID[*omni.Machine](ctx, r, machine)
-		if err != nil && state.IsNotFoundError(err) {
-			return true, nil
-		}
-
+	if err != nil {
 		return false, err
 	}
 
-	// The order is important here - ClusterMachine and MachineSet patches also has
-	// the Cluster label set, so the Cluster label should be checked last.
-
-	if clusterMachine, ok := configPatch.Metadata().Labels().Get(omni.LabelClusterMachine); ok {
-		_, err := safe.ReaderGetByID[*omni.ClusterMachine](ctx, r, clusterMachine)
-		if err != nil && state.IsNotFoundError(err) {
-			return true, nil
-		}
+	return false, nil
+}
 
-		return false, err
+// isOrphan checks if the ConfigPatch is orphaned.
+func (ctrl *ConfigPatchCleanupController) isOrphan(ctx context.Context, r controller.Reader, configPatch *omni.ConfigPatch) (bool, error) {
+	if configPatch.Metadata().Owner() != "" {
+		return false, nil
 	}
 
-	if machineSet, ok := configPatch.Metadata().Labels().Get(omni.LabelMachineSet); ok {
-		_, err := safe.ReaderGetByID[*omni.MachineSet](ctx, r, machineSet)
-		if err != nil && state.IsNotFoundError(err) {
-			return true, nil
-		}
-
-		return false, err
+	if time.Since(configPatch.Metadata().Updated()) < ctrl.DeleteOlderThan {
+		return false, nil
 	}
 
-	if cluster, ok := configPatch.Metadata().Labels().Get(omni.LabelCluster); ok {
-		_, err := safe.ReaderGetByID[*omni.Cluster](ctx, r, cluster)
-		if err != nil && state.IsNotFoundError(err) {
-			return true, nil
+	for _, labelCheck := range []func() (bool, error){
```

Comment:
this changes the logic - a patch is orphaned only if it's orphaned on all labels (not a single label like before)